subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt

```diff
@@ -1,47 +1,229 @@
-import com.avito.runner.config.InstrumentationTestsActionParams
+import com.avito.android.runner.report.ReportFactoryImpl
+import com.avito.android.stats.SeriesName
+import com.avito.android.stats.StatsDSender
+import com.avito.http.HttpClientProvider
+import com.avito.logger.LoggerFactory
+import com.avito.runner.config.RunnerInputParams
+import com.avito.runner.finalizer.FinalizerFactory
+import com.avito.runner.finalizer.FinalizerFactoryImpl
 import com.avito.runner.scheduler.TestRunnerFactoryProvider
 import com.avito.runner.scheduler.suite.TestSuiteProvider
 import com.avito.runner.scheduler.suite.filter.FilterFactory
 import com.avito.runner.scheduler.suite.filter.FilterInfoWriter
+import com.avito.runner.service.worker.device.adb.listener.RunnerMetricsConfig
+import com.avito.time.DefaultTimeProvider
+import com.avito.time.TimeProvider
 
-public class TestSchedulerFactoryImpl(
-    private val params: InstrumentationTestsActionParams,
-    private val report: Report,
-    private val testSuiteLoader: TestSuiteLoader,
-    private val reportFactory: ReportFactory,
-    private val testRunnerFactoryProvider: TestRunnerFactoryProvider,
-) : TestSchedulerFactory {
+internal class TestSchedulerFactoryImpl : TestSchedulerFactory {
 
-    override fun create(): TestScheduler {
-        val testSuiteProvider: TestSuiteProvider = createTestSuiteProvider()
+    override fun create(params: RunnerInputParams): TestScheduler {
+
+        val timeProvider: TimeProvider = DefaultTimeProvider()
+
+        val statsdSender = createStatsdSender(params)
+
+        val httpClientProvider = HttpClientProvider(
+            statsDSender = statsdSender,
+            timeProvider = timeProvider,
+            loggerFactory = params.loggerFactory
+        )
+
+        val reportFactory: ReportFactory = createReportFactory(
+            params = params,
+            timeProvider = timeProvider,
+            httpClientProvider = httpClientProvider
+        )
+
+        val report = reportFactory.createReport()
+
+        val filterFactory = createFilterFactory(
+            params = params,
+            reportFactory = reportFactory
+        )
+
+        val testSuiteProvider: TestSuiteProvider = createTestSuiteProvider(
+            params = params,
+            report = report,
+            filterFactory = filterFactory
+        )
+
+        val metricsConfig: RunnerMetricsConfig = createRunnerMetricsConfig(params)
+
+        val finalizerFactory: FinalizerFactory = createFinalizerFactory(
+            params = params,
+            metricsConfig = metricsConfig,
+            reportFactory = reportFactory,
+            timeProvider = timeProvider
+        )
+
+        val androidDebugBridgeProvider = createAndroidDebugBridgeProvider(params.loggerFactory)
+
+        val emulatorsLogsReporterProvider = createEmulatorsLogsReporterProvider(params)
+
+        val testRunnerFactoryProvider = createTestRunnerFactoryProvider(
```

Comment:
I think that is not what `TestShecdulerFactory` should do
`TestSchedulerFactory` must only create the `TestScheduler` and get `testRunnerFactoryProvider` as an input parameter

Reply:
so we need something like TestSchedulerFactoryFactory, or entrypoint for runner?